src/libs/antares/study/parts/thermal/cluster_list.cpp

```diff
@@ -159,52 +159,57 @@ bool ThermalClusterList::loadFromFolder(Study& study, const AnyString& folder, A
             options = Matrix<>::optFixedSize,
         };
 
-        bool r = cluster->modulation.loadFromCSVFile(modulationFile,
-                                                     thermalModulationMax,
-                                                     HOURS_PER_YEAR,
-                                                     options);
-        if (!r && study.usedByTheSolver)
+        ret = cluster->modulation.loadFromCSVFile(modulationFile,
+                                                  thermalModulationMax,
+                                                  HOURS_PER_YEAR,
+                                                  options) && ret;
+
+        // Check the data integrity of the cluster
+        addToCompleteList(cluster);
+    }
+
+    rebuildIndexes();
+    rebuildIndex();
+
+    return ret;
+}
+
+
+bool ThermalClusterList::validateClusters(const Parameters& parameters) const
+{
+    bool ret = true;
+
+    for (const auto& cluster : allClusters_)
+    {
+        // update the minUpDownTime
+        cluster->minUpDownTime = std::max(cluster->minUpTime, cluster->minDownTime);
```

Comment:
Need a bit of clarification, maybe more details in the comment. It's strange to see a std::max on a minX value.
Let's add information on why we're doing this

Reply:
```
    //! Max entre . minUp/minDown time (1..168)
    uint minUpDownTime;
```
The naming isn't great, can be changed to maxMinUpMinDownTime